src/coreclr/gc/gc.cpp

```diff
@@ -11662,6 +11668,17 @@ void gc_heap::set_region_plan_gen_num (heap_segment* region, int plan_gen_num)
         region->flags &= ~heap_segment_flags_demoted;
     }
 
+    // If replace_p is true, it means we need to move a region from its original planned gen to this new gen.
+    if (replace_p)
+    {
+        int original_plan_gen_num = heap_segment_plan_gen_num (region);
+        planned_regions_per_gen[original_plan_gen_num]--;
+    }
+
+    planned_regions_per_gen[plan_gen_num]++;
+    dprintf (REGIONS_LOG, ("h%d g%d %Ix(%Ix) -> g%d (total %d region planned in g%d)",
```

Comment:
I wonder if it would be useful to add the `original_plan_gen_num` to this dprintf as well.

Nit: I thought we switched the `%Ix` formats to `%zx`?

Reply:
fixed